src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksDtoToPaymentWorksVendorConversionServiceImpl.java

```diff
@@ -0,0 +1,246 @@
+            populateNewVendorBankAddressAttributes(stgNewVendor, pmwBankAccountDTO);
+        }
+    }
+
+    private void populateNewVendorBankAddressAttributes(PaymentWorksVendor stgNewVendor, PaymentWorksBankAccountDTO pmwBankAccountDTO) {
+        if (bankAddressExists(pmwBankAccountDTO)) {
+            PaymentWorksBankAddressDTO pmwBankAddressDTO = pmwBankAccountDTO.getBank_address();
+            stgNewVendor.setBankAddressStreet1(pmwBankAddressDTO.getStreet1());
+            stgNewVendor.setBankAddressStreet2(pmwBankAddressDTO.getStreet2());
+            stgNewVendor.setBankAddressCity(pmwBankAddressDTO.getCity());
+            stgNewVendor.setBankAddressState(pmwBankAddressDTO.getState());
+            stgNewVendor.setBankAddressCountry(pmwBankAddressDTO.getCountry());
+            stgNewVendor.setBankAddressZipCode(pmwBankAddressDTO.getZipcode());
+            stgNewVendor.setBankAddressValidated(pmwBankAddressDTO.getValidated());
+        }
+    }
+
+    private boolean newVendorDetailExists(PaymentWorksNewVendorRequestDetailDTO paymentWorksNewVendorDetailDTO) {
+        return (ObjectUtils.isNotNull(paymentWorksNewVendorDetailDTO));
+    }
+    
+    private boolean requestingCompanyExists(PaymentWorksNewVendorRequestDetailDTO paymentWorksNewVendorDetailDTO) {
+        return (ObjectUtils.isNotNull(paymentWorksNewVendorDetailDTO.getRequesting_company()));
+    }
+    
+    private boolean singleRemittanceAddressExists(PaymentWorksRequestingCompanyDTO pmwRequestingCompanyDTO) {
+        return (ObjectUtils.isNotNull(pmwRequestingCompanyDTO.getRemittance_addresses()) &&
+                ObjectUtils.isNotNull(pmwRequestingCompanyDTO.getRemittance_addresses().getRemittance_address()) &&
+                pmwRequestingCompanyDTO.getRemittance_addresses().getRemittance_address().size() == 1);
+    }
+
+    private boolean corporateAddressExists(PaymentWorksRequestingCompanyDTO pmwRequestingCompanyDTO) {
+        return (ObjectUtils.isNotNull(pmwRequestingCompanyDTO.getCorp_address()));
+    }
+
+    private boolean bankAccountDataExists(PaymentWorksRemittanceAddressDTO pmwRemittanceAddressDTO) {
+        return (ObjectUtils.isNotNull(pmwRemittanceAddressDTO.getBank_acct()));
+    }
+
+    private boolean bankAddressExists(PaymentWorksBankAccountDTO pmwBankAccountDTO) {
+        return (ObjectUtils.isNotNull(pmwBankAccountDTO.getBank_address()));
+    }
+    
+    private boolean taxClassificationExists(PaymentWorksRequestingCompanyDTO pmwRequestingCompanyDTO) {
+        return (ObjectUtils.isNotNull(pmwRequestingCompanyDTO.getTax_classification()));
+    }
+    
+    private void extractCustomFields(PaymentWorksVendor stgNewVendor, PaymentWorksNewVendorRequestDetailDTO pmwNewVendorDetailDTO, PaymentWorksNewVendorRequestsBatchReportData reportData) {
+        Map<String, String> customFieldsDTOMap = createCustomFieldIdCustomFieldValueMap(pmwNewVendorDetailDTO.getCustom_fields());
```

Comment:
this method is quite complicated, it could probably be broken down into smaller units

Reply:
Broke method up as requested.